File: src/main/java/javabot/commands/AddAdmin.java
package javabot.commands;

import com.antwerkz.sofia.Sofia;
import com.google.inject.Provider;
import javabot.Message;
import javabot.dao.AdminDao;
import org.pircbotx.PircBotX;
import org.pircbotx.User;

import javax.inject.Inject;

public class AddAdmin extends AdminCommand {
    @Inject
    private AdminDao dao;
    @Inject
    private Provider<PircBotX> ircBot;
    @Param
    String userName;
    @Param
    String hostName;

    @Override
    public void execute(final Message event) {
        final User user = findUser(userName);
        if (user == null) {
            getJavabot().postMessageToChannel(event, Sofia.userNotFound(userName));
        } else {
            if (dao.getAdmin(user.getNick(), hostName) != null) {
                getJavabot().postMessageToChannel(event, Sofia.adminAlready(user.getNick()));
            } else {
                dao.create(user.getNick(), user.getLogin(), user.getHostmask());
                getJavabot().postMessageToChannel(event, Sofia.adminAdded(user.getNick()));
            }
        }
    }

    public User findUser(final String name) {
        return ircBot.get().getUserChannelDao().getUser(name);
    }

}

File: src/main/kotlin/javabot/commands/AddAdmin.kt
package javabot.commands

import com.antwerkz.sofia.Sofia
import com.beust.jcommander.Parameter
import javabot.Javabot
import javabot.Message
import javabot.dao.AdminDao
import javax.inject.Inject

class AddAdmin @Inject constructor(bot: Javabot, adminDao: AdminDao) :
        AdminCommand(bot, adminDao) {
    @Parameter(required = true)
    lateinit var userName: String
    @Parameter(required = true)
    lateinit var hostName: String

    override fun execute(event: Message): List<Message> {
        val responses = arrayListOf<Message>()
        val user = event.user
        if (adminDao.getAdmin(user.nick, hostName) != null) {
            responses.add(Message(event, Sofia.adminAlready(user.nick)))
        } else {
            adminDao.create(user.nick, user.userName, user.hostmask)
            responses.add(Message(event, Sofia.adminAdded(user.nick)))
        }

        return responses
    }
}
